aspnet/fundamentals/localization.rst

```diff
@@ -1,8 +1,285 @@
 
-|stub-icon| Localization
-========================
+By `Rick Anderson`_ with assistance from `Damien Bowden <https://twitter.com/damien_bod>`_, `Bart Calixto <https://twitter.com/bartmax>`_, and `Nadeem Afana <https://twitter.com/NadeemAfana>`__ 
 
-.. include:: /../common/stub-notice.txt
+Creating a multilingual website with ASP.NET Core will allow your site to reach a wider audience. ASP.NET has :doc:`/fundamentals/request-features` that provide support for localizing into different languages and cultures.
 
-.. _issue: https://github.com/aspnet/Docs/issues/653
\ No newline at end of file
+This sample is based on the **Localization.StarterWeb** project on `GitHub <https://github.com/aspnet/entropy>`__.
+
+.. contents:: In this article:
+  :local:
+  :depth: 1
+  
+.. note:: Currently, resource files are not read when the project is run from Visual Studio. See `this issue <https://github.com/aspnet/dnx/issues/3047>`_ for more information. Until the issue with Visual Studio is addressed, you can test the project by running it from the command line.
+
+IStringLocalizer
+---------------------------------
+
+Introduced in ASP.NET Core, ``IStringLocalizer`` was architected to improve productivity when developing localized apps. 
+
+The simple interface has an indexer and an ``IEnumerable`` for returning localized strings. Unlike previous localization approaches, ``IStringLocalizer`` doesn't require you to store the default language strings in a resource file. You can develop an app targeted for localization and not need to create resource files early in development. We anticipate that many developers will **not** create a default language resource file. In previous versions of ASP.NET, you had to move all your localization strings into a resource (.resx) file and change your app everywhere the strings were used, even for the default language. You no longer need to move your localized strings, you simply wrap them.
+
+.. literalinclude:: localization/sample/Controllers/AboutController.cs
+  :language: c#
+
+In the code above, the ``IStringLocalizer`` implementation comes from :doc:`/fundamentals/dependency-injection`. If the localized value of "About Title" is not found, then the indexer key is returned, that is, the string "About Title". You can leave the default language literal strings in the app and wrap them in the localizer. You can develop your app with your default language and prepare it for the localization step without first creating a default resource file. Alternatively, you can use the traditional approach and provide a key to retrieve the default language string. For many developers the new workflow of not having a default language "resx" file and simply wrapping the string literals can reduce the overhead of localizing an app. Other developers will prefer the traditional work flow as it can make it easier to work with longer string literals and make it easier to update localized strings.
```

Comment:
@danroth27 I changed it to 
I'll show how the `IStringLocalizer` service gets added in the **Configuring localization** section below.
I can't move only the **Configuring localization** section before this, as the **Localization middleware** section needs to follow **Configuring localization** - and that makes for way too much detail before we get to the most interesting part (AKA `IStringLocalizer` ). That's why I originally move the globalization/internationalization/localization section to the end. Now that all that's in the intro, it's a long way to `IStringLocalizer`. I think my current approach is better - but I'm will to move it up config/middleware if you think that approach is better.